components/ResourceCreation/TestResourcesDisplay.tsx

```diff
@@ -5,43 +5,101 @@ import { patientTestCaseState } from '../../state/atoms/patientTestCase';
+import { DataRequirement } from 'fhir/r4';
+
+interface DataSearchKeys {
+  type: string;
+  id: string | undefined;
+  displayString: string;
+}
 
 export default function TestResourcesDisplay() {
   const dataRequirements = useRecoilValue(dataRequirementsState);
   const [, setSelectedDataRequirement] = useRecoilState(selectedDataRequirementState);
   const valueSetMap = useRecoilValue(valueSetMapState);
   const currentPatients = useRecoilValue(patientTestCaseState);
+  const searchQuery = useRecoilValue(searchQueryState);
+  const [dataReqsToDisplay, setDataReqsToDisplay] = useState<DataRequirement[] | null>(dataRequirements);
 
-  return dataRequirements?.length && Object.keys(currentPatients).length ? (
-    <div
-      data-testid="test-resource-panel"
-      style={{
-        maxHeight: '40vh',
-        overflow: 'scroll'
-      }}
-    >
-      <Stack>
-        {dataRequirements.map((dr, i) => {
-          const key = `${dr.type}${i}`;
+  useEffect(() => {
+    /**
+     * Assembles an array of data that will be searched by FuseJS to narrow
+     * down the displayed data requirements in the test resources display.
+     * @returns array of data to use for fuzzy searching
+     */
+    const getSearchContent = (): DataSearchKeys[] | undefined => {
+      if (dataRequirements) {
+        const searchableData = dataRequirements.map(dr => {
+          const type = dr.type;
+          const id = dr.id;
           const displayString = getDataRequirementFiltersString(dr, valueSetMap);
-          return (
-            <Button
-              key={key}
-              onClick={() => {
-                setSelectedDataRequirement({ name: displayString, content: dr });
-              }}
-              fullWidth
-              variant="default"
-              style={{ height: 'auto', padding: 10, overflow: 'hidden' }}
-            >
-              <div style={{ overflow: 'hidden' }}>
-                <Title order={3}>{dr.type}</Title>
-                <Text style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>{displayString}</Text>
-              </div>
-            </Button>
-          );
-        })}
-      </Stack>
-    </div>
+          return { type, id, displayString };
+        });
+        return searchableData;
+      }
+    };
+
+    /**
+     * Uses FuseJS to do a fuzzy search on the search contents. Then, filters the
+     * data requirements so that the displayed data requirements are those that
+     * came back from the search.
+     * @returns array of data requirements returned from FuseJS search
+     */
+    const filterBySearchResults = () => {
+      const searchContents = getSearchContent();
+      if (dataRequirements && searchContents) {
+        const keys: (keyof DataSearchKeys)[] = ['type', 'id', 'displayString'];
+        const fuse = new Fuse(searchContents, { keys: keys });
+        const results = fuse.search(searchQuery).map(dr => dr.item.displayString);
+        const foundDrs = dataRequirements.filter(dr => {
```

Comment:
If I understand what this is doing, this is going through the existing data requirements array, filtering it down to only include ones that exist in the search results returned by fuse, and then using those as what to display. This is good, except I think it's missing one crucial piece of information.

By default, the list of results that Fuse returns is [sorted](https://fusejs.io/api/options.html#shouldsort) according to how well that result matches up with the search query. It would be nice if that order is preserved when we're rending the data requirements, that way we can have the most relevant results at the top. This is a little tricky though, given that we need access to the whole data requirement object in order to do the iteration. Maybe we can try to brainstorm a way to do this that isn't `O(n^2)`. It might be hard but I think it'd be nice if we could pull that off

Reply:
One potential solution to this is to index the entire data requirement in the search index, but still include those other properties and only search by those, but then we can use the resulting dataRequirement off of the returned search items

Reply:
Good idea, Matt!

Reply:
Thanks

Reply:
@mgramigna I think this approach will definitely work, because we can just pull the data requirement off of the `.item` property of the search results once they get returned!